File: mRemoteNG/Themes/ThemeInfo.vb
Imports System.ComponentModel
Imports mRemoteNG.Tools.LocalizedAttributes
Imports mRemoteNG.My

Namespace Themes
    Public Class ThemeInfo
        Implements ICloneable, INotifyPropertyChanged
#Region "Public Methods"
        Public Sub New(Optional ByVal themeName As String = Nothing)
            If themeName IsNot Nothing Then Name = themeName
        End Sub

        Public Function Clone() As Object Implements ICloneable.Clone
            Return MemberwiseClone()
        End Function

        Public Overrides Function ToString() As String
            Return Name
        End Function

        Public Overrides Function Equals(obj As Object) As Boolean
            Dim otherTheme As ThemeInfo = TryCast(obj, ThemeInfo)
            If otherTheme Is Nothing Then Return False

            Dim themeInfoType As Type = (New ThemeInfo).GetType()
            Dim myProperty As Object
            Dim otherProperty As Object
            For Each propertyInfo As Reflection.PropertyInfo In themeInfoType.GetProperties()
                myProperty = propertyInfo.GetValue(Me, Nothing)
                otherProperty = propertyInfo.GetValue(otherTheme, Nothing)
                If Not myProperty.Equals(otherProperty) Then Return False
            Next

            Return True
        End Function
#End Region

#Region "Events"
        Public Event PropertyChanged(sender As Object, e As System.ComponentModel.PropertyChangedEventArgs) Implements System.ComponentModel.INotifyPropertyChanged.PropertyChanged
        Protected Sub NotifyPropertyChanged(ByVal propertyName As String)
            RaiseEvent PropertyChanged(Me, New PropertyChangedEventArgs(propertyName))
        End Sub
#End Region

#Region "Properties"
        Private _name As String = Language.strUnnamedTheme
        <Browsable(False)> _
        Public Property Name As String
            Get
                Return _name
            End Get
            Set(value As String)
                If _name = value Then Return
                _name = value
                NotifyPropertyChanged("Name")
            End Set
        End Property

#Region "General"
        Private _windowBackgroundColor As Color = SystemColors.AppWorkspace
        <LocalizedCategory("strThemeCategoryGeneral", 1), _
            LocalizedDisplayName("strThemeNameWindowBackgroundColor"), _
            LocalizedDescription("strThemeDescriptionWindowBackgroundColor")> _
        Public Property WindowBackgroundColor() As Color
            Get
                Return (_windowBackgroundColor)
            End Get
            Set(value As Color)
                If _windowBackgroundColor = value Then Return
                _windowBackgroundColor = value
                NotifyPropertyChanged("WindowBackgroundColor")
            End Set
        End Property

        Private _menuBackgroundColor As Color = SystemColors.Control
        <LocalizedCategory("strThemeCategoryGeneral", 1), _
            Browsable(False),
            LocalizedDisplayName("strThemeNameMenuBackgroundColor"),
            LocalizedDescription("strThemeDescriptionMenuBackgroundColor")>
        Public Property MenuBackgroundColor() As Color
            Get
                Return _menuBackgroundColor
            End Get
            Set(value As Color)
                If _menuBackgroundColor = value Then Return
                _menuBackgroundColor = value
                NotifyPropertyChanged("MenuBackgroundColor")
            End Set
        End Property

        Private _menuTextColor As Color = SystemColors.ControlText
        <LocalizedCategory("strThemeCategoryGeneral", 1), _
            Browsable(False),
            LocalizedDisplayName("strThemeNameMenuTextColor"),
            LocalizedDescription("strThemeDescriptionMenuTextColor")>
        Public Property MenuTextColor() As Color
            Get
                Return _menuTextColor
            End Get
            Set(value As Color)
                If _menuTextColor = value Then Return
                _menuTextColor = value
                NotifyPropertyChanged("MenuTextColor")
            End Set
        End Property

        Private _toolbarBackgroundColor As Color = SystemColors.Control
        <LocalizedCategory("strThemeCategoryGeneral", 1), _
            Browsable(False), _
            LocalizedDisplayName("strThemeNameToolbarBackgroundColor"), _
            LocalizedDescription("strThemeDescriptionToolbarBackgroundColor")> _
        Public Property ToolbarBackgroundColor() As Color
            Get
                Return _toolbarBackgroundColor
            End Get
            Set(value As Color)
                If _toolbarBackgroundColor = value Or value.A < 255 Then Return
                _toolbarBackgroundColor = value
                NotifyPropertyChanged("ToolbarBackgroundColor")
            End Set
        End Property

        Private _toolbarTextColor As Color = SystemColors.ControlText
        <LocalizedCategory("strThemeCategoryGeneral", 1), _
            Browsable(False), _
            LocalizedDisplayName("strThemeNameToolbarTextColor"), _
            LocalizedDescription("strThemeDescriptionToolbarTextColor")> _
        Public Property ToolbarTextColor() As Color
            Get
                Return _toolbarTextColor
            End Get
            Set(value As Color)
                If _toolbarTextColor = value Then Return
                _toolbarTextColor = value
                NotifyPropertyChanged("ToolbarTextColor")
            End Set
        End Property
#End Region

#Region "Connections Panel"
        Private _connectionsPanelBackgroundColor As Color = SystemColors.Window
        <LocalizedCategory("strThemeCategoryConnectionsPanel", 2), _
            LocalizedDisplayName("strThemeNameConnectionsPanelBackgroundColor"), _
            LocalizedDescription("strThemeDescriptionConnectionsPanelBackgroundColor")> _
        Public Property ConnectionsPanelBackgroundColor() As Color
            Get
                Return _connectionsPanelBackgroundColor
            End Get
            Set(value As Color)
                If _connectionsPanelBackgroundColor = value Or value.A < 255 Then Return
                _connectionsPanelBackgroundColor = value
                NotifyPropertyChanged("ConnectionsPanelBackgroundColor")
            End Set
        End Property

        Private _connectionsPanelTextColor As Color = SystemColors.WindowText
        <LocalizedCategory("strThemeCategoryConnectionsPanel", 2),
            LocalizedDisplayName("strThemeNameConnectionsPanelTextColor"),
            LocalizedDescription("strThemeDescriptionConnectionsPanelTextColor")>
        Public Property ConnectionsPanelTextColor() As Color
            Get
                Return _connectionsPanelTextColor
            End Get
            Set(value As Color)
                If _connectionsPanelTextColor = value Then Return
                _connectionsPanelTextColor = value
                NotifyPropertyChanged("ConnectionsPanelTextColor")
            End Set
        End Property

        Private _connectionsPanelTreeLineColor As Color = Color.Black
        <LocalizedCategory("strThemeCategoryConnectionsPanel", 2),
            LocalizedDisplayName("strThemeNameConnectionsPanelTreeLineColor"),
            LocalizedDescription("strThemeDescriptionConnectionsPanelTreeLineColor")>
        Public Property ConnectionsPanelTreeLineColor() As Color
            Get
                Return _connectionsPanelTreeLineColor
            End Get
            Set(value As Color)
                If _connectionsPanelTreeLineColor = value Then Return
                _connectionsPanelTreeLineColor = value
                NotifyPropertyChanged("ConnectionsPanelTreeLineColor")
            End Set
        End Property

        Private _searchBoxBackgroundColor As Color = SystemColors.Window
        <LocalizedCategory("strThemeCategoryConnectionsPanel", 2), _
            LocalizedDisplayName("strThemeNameSearchBoxBackgroundColor"), _
            LocalizedDescription("strThemeDescriptionSearchBoxBackgroundColor")> _
        Public Property SearchBoxBackgroundColor() As Color
            Get
                Return _searchBoxBackgroundColor
            End Get
            Set(value As Color)
                If _searchBoxBackgroundColor = value Or value.A < 255 Then Return
                _searchBoxBackgroundColor = value
                NotifyPropertyChanged("SearchBoxBackgroundColor")
            End Set
        End Property

        Private _searchBoxTextPromptColor As Color = SystemColors.GrayText
        <LocalizedCategory("strThemeCategoryConnectionsPanel", 2), _
            LocalizedDisplayName("strThemeNameSearchBoxTextPromptColor"), _
            LocalizedDescription("strThemeDescriptionSearchBoxTextPromptColor")> _
        Public Property SearchBoxTextPromptColor() As Color
            Get
                Return _searchBoxTextPromptColor
            End Get
            Set(value As Color)
                If _searchBoxTextPromptColor = value Then Return
                _searchBoxTextPromptColor = value
                NotifyPropertyChanged("SearchBoxTextPromptColor")
            End Set
        End Property

        Private _searchBoxTextColor As Color = SystemColors.WindowText
        <LocalizedCategory("strThemeCategoryConnectionsPanel", 2), _
            LocalizedDisplayName("strThemeNameSearchBoxTextColor"), _
            LocalizedDescription("strThemeDescriptionSearchBoxTextColor")> _
        Public Property SearchBoxTextColor() As Color
            Get
                Return _searchBoxTextColor
            End Get
            Set(value As Color)
                If _searchBoxTextColor = value Then Return
                _searchBoxTextColor = value
                NotifyPropertyChanged("SearchBoxTextColor")
            End Set
        End Property
#End Region

#Region "Config Panel"
        Private _configPanelBackgroundColor As Color = SystemColors.Window
        <LocalizedCategory("strThemeCategoryConfigPanel", 3), _
            LocalizedDisplayName("strThemeNameConfigPanelBackgroundColor"), _
            LocalizedDescription("strThemeDescriptionConfigPanelBackgroundColor")> _
        Public Property ConfigPanelBackgroundColor() As Color
            Get
                Return _configPanelBackgroundColor
            End Get
            Set(value As Color)
                If _configPanelBackgroundColor = value Or value.A < 255 Then Return
                _configPanelBackgroundColor = value
                NotifyPropertyChanged("ConfigPanelBackgroundColor")
            End Set
        End Property

        Private _configPanelTextColor As Color = SystemColors.WindowText
        <LocalizedCategory("strThemeCategoryConfigPanel", 3), _
            LocalizedDisplayName("strThemeNameConfigPanelTextColor"), _
            LocalizedDescription("strThemeDescriptionConfigPanelTextColor")> _
        Public Property ConfigPanelTextColor() As Color
            Get
                Return _configPanelTextColor
            End Get
            Set(value As Color)
                If _configPanelTextColor = value Then Return
                _configPanelTextColor = value
                NotifyPropertyChanged("ConfigPanelTextColor")
            End Set
        End Property

        Private _configPanelCategoryTextColor As Color = SystemColors.ControlText
        <LocalizedCategory("strThemeCategoryConfigPanel", 3), _
            LocalizedDisplayName("strThemeNameConfigPanelCategoryTextColor"), _
            LocalizedDescription("strThemeDescriptionConfigPanelCategoryTextColor")> _
        Public Property ConfigPanelCategoryTextColor() As Color
            Get
                Return _configPanelCategoryTextColor
            End Get
            Set(value As Color)
                If _configPanelCategoryTextColor = value Then Return
                _configPanelCategoryTextColor = value
                NotifyPropertyChanged("ConfigPanelCategoryTextColor")
            End Set
        End Property

        Private _configPanelHelpBackgroundColor As Color = SystemColors.Control
        <LocalizedCategory("strThemeCategoryConfigPanel", 3), _
            LocalizedDisplayName("strThemeNameConfigPanelHelpBackgroundColor"), _
            LocalizedDescription("strThemeDescriptionConfigPanelHelpBackgroundColor")> _
        Public Property ConfigPanelHelpBackgroundColor() As Color
            Get
                Return _configPanelHelpBackgroundColor
            End Get
            Set(value As Color)
                If _configPanelHelpBackgroundColor = value Or value.A < 255 Then Return
                _configPanelHelpBackgroundColor = value
                NotifyPropertyChanged("ConfigPanelHelpBackgroundColor")
            End Set
        End Property

        Private _configPanelHelpTextColor As Color = SystemColors.ControlText
        <LocalizedCategory("strThemeCategoryConfigPanel", 3), _
            LocalizedDisplayName("strThemeNameConfigPanelHelpTextColor"), _
            LocalizedDescription("strThemeDescriptionConfigPanelHelpTextColor")> _
        Public Property ConfigPanelHelpTextColor() As Color
            Get
                Return _configPanelHelpTextColor
            End Get
            Set(value As Color)
                If _configPanelHelpTextColor = value Then Return
                _configPanelHelpTextColor = value
                NotifyPropertyChanged("ConfigPanelHelpTextColor")
            End Set
        End Property

        Private _configPanelGridLineColor As Color = SystemColors.InactiveBorder
        <LocalizedCategory("strThemeCategoryConfigPanel", 3), _
            LocalizedDisplayName("strThemeNameConfigPanelGridLineColor"), _
            LocalizedDescription("strThemeDescriptionConfigPanelGridLineColor")> _
        Public Property ConfigPanelGridLineColor() As Color
            Get
                Return _configPanelGridLineColor
            End Get
            Set(value As Color)
                If _configPanelGridLineColor = value Then Return
                _configPanelGridLineColor = value
                NotifyPropertyChanged("ConfigPanelGridLineColor")
            End Set
        End Property
#End Region
#End Region
    End Class
End Namespace

File: mRemoteV1.ConvertedToC#/Themes/ThemeInfo.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;
using System.ComponentModel;
using mRemoteNG.Tools.LocalizedAttributes;
using mRemoteNG.My;

namespace mRemoteNG.Themes
{
	public class ThemeInfo : ICloneable, INotifyPropertyChanged
	{
		#region "Public Methods"
		public ThemeInfo(string themeName = null)
		{
			if (themeName != null)
				Name = themeName;
		}

		public object Clone()
		{
			return MemberwiseClone();
		}

		public override string ToString()
		{
			return Name;
		}

		public override bool Equals(object obj)
		{
			ThemeInfo otherTheme = obj as ThemeInfo;
			if (otherTheme == null)
				return false;

			Type themeInfoType = (new ThemeInfo()).GetType();
			object myProperty = null;
			object otherProperty = null;
			foreach (System.Reflection.PropertyInfo propertyInfo in themeInfoType.GetProperties()) {
				myProperty = propertyInfo.GetValue(this, null);
				otherProperty = propertyInfo.GetValue(otherTheme, null);
				if (!myProperty.Equals(otherProperty))
					return false;
			}

			return true;
		}
		#endregion

		#region "Events"
		public event PropertyChangedEventHandler System.ComponentModel.INotifyPropertyChanged.PropertyChanged;
		public delegate void PropertyChangedEventHandler(object sender, System.ComponentModel.PropertyChangedEventArgs e);
		protected void NotifyPropertyChanged(string propertyName)
		{
			if (PropertyChanged != null) {
				PropertyChanged(this, new PropertyChangedEventArgs(propertyName));
			}
		}
		#endregion

		#region "Properties"
		private string _name = Language.strUnnamedTheme;
		[Browsable(false)]
		public string Name {
			get { return _name; }
			set {
				if (_name == value)
					return;
				_name = value;
				NotifyPropertyChanged("Name");
			}
		}

		#region "General"
		private Color _windowBackgroundColor = SystemColors.AppWorkspace;
		[LocalizedCategory("strThemeCategoryGeneral", 1), LocalizedDisplayName("strThemeNameWindowBackgroundColor"), LocalizedDescription("strThemeDescriptionWindowBackgroundColor")]
		public Color WindowBackgroundColor {
			get { return (_windowBackgroundColor); }
			set {
				if (_windowBackgroundColor == value)
					return;
				_windowBackgroundColor = value;
				NotifyPropertyChanged("WindowBackgroundColor");
			}
		}

		private Color _menuBackgroundColor = SystemColors.Control;
		[LocalizedCategory("strThemeCategoryGeneral", 1), Browsable(false), LocalizedDisplayName("strThemeNameMenuBackgroundColor"), LocalizedDescription("strThemeDescriptionMenuBackgroundColor")]
		public Color MenuBackgroundColor {
			get { return _menuBackgroundColor; }
			set {
				if (_menuBackgroundColor == value)
					return;
				_menuBackgroundColor = value;
				NotifyPropertyChanged("MenuBackgroundColor");
			}
		}

		private Color _menuTextColor = SystemColors.ControlText;
		[LocalizedCategory("strThemeCategoryGeneral", 1), Browsable(false), LocalizedDisplayName("strThemeNameMenuTextColor"), LocalizedDescription("strThemeDescriptionMenuTextColor")]
		public Color MenuTextColor {
			get { return _menuTextColor; }
			set {
				if (_menuTextColor == value)
					return;
				_menuTextColor = value;
				NotifyPropertyChanged("MenuTextColor");
			}
		}

		private Color _toolbarBackgroundColor = SystemColors.Control;
		[LocalizedCategory("strThemeCategoryGeneral", 1), Browsable(false), LocalizedDisplayName("strThemeNameToolbarBackgroundColor"), LocalizedDescription("strThemeDescriptionToolbarBackgroundColor")]
		public Color ToolbarBackgroundColor {
			get { return _toolbarBackgroundColor; }
			set {
				if (_toolbarBackgroundColor == value | value.A < 255)
					return;
				_toolbarBackgroundColor = value;
				NotifyPropertyChanged("ToolbarBackgroundColor");
			}
		}

		private Color _toolbarTextColor = SystemColors.ControlText;
		[LocalizedCategory("strThemeCategoryGeneral", 1), Browsable(false), LocalizedDisplayName("strThemeNameToolbarTextColor"), LocalizedDescription("strThemeDescriptionToolbarTextColor")]
		public Color ToolbarTextColor {
			get { return _toolbarTextColor; }
			set {
				if (_toolbarTextColor == value)
					return;
				_toolbarTextColor = value;
				NotifyPropertyChanged("ToolbarTextColor");
			}
		}
		#endregion

		#region "Connections Panel"
		private Color _connectionsPanelBackgroundColor = SystemColors.Window;
		[LocalizedCategory("strThemeCategoryConnectionsPanel", 2), LocalizedDisplayName("strThemeNameConnectionsPanelBackgroundColor"), LocalizedDescription("strThemeDescriptionConnectionsPanelBackgroundColor")]
		public Color ConnectionsPanelBackgroundColor {
			get { return _connectionsPanelBackgroundColor; }
			set {
				if (_connectionsPanelBackgroundColor == value | value.A < 255)
					return;
				_connectionsPanelBackgroundColor = value;
				NotifyPropertyChanged("ConnectionsPanelBackgroundColor");
			}
		}

		private Color _connectionsPanelTextColor = SystemColors.WindowText;
		[LocalizedCategory("strThemeCategoryConnectionsPanel", 2), LocalizedDisplayName("strThemeNameConnectionsPanelTextColor"), LocalizedDescription("strThemeDescriptionConnectionsPanelTextColor")]
		public Color ConnectionsPanelTextColor {
			get { return _connectionsPanelTextColor; }
			set {
				if (_connectionsPanelTextColor == value)
					return;
				_connectionsPanelTextColor = value;
				NotifyPropertyChanged("ConnectionsPanelTextColor");
			}
		}

		private Color _connectionsPanelTreeLineColor = Color.Black;
		[LocalizedCategory("strThemeCategoryConnectionsPanel", 2), LocalizedDisplayName("strThemeNameConnectionsPanelTreeLineColor"), LocalizedDescription("strThemeDescriptionConnectionsPanelTreeLineColor")]
		public Color ConnectionsPanelTreeLineColor {
			get { return _connectionsPanelTreeLineColor; }
			set {
				if (_connectionsPanelTreeLineColor == value)
					return;
				_connectionsPanelTreeLineColor = value;
				NotifyPropertyChanged("ConnectionsPanelTreeLineColor");
			}
		}

		private Color _searchBoxBackgroundColor = SystemColors.Window;
		[LocalizedCategory("strThemeCategoryConnectionsPanel", 2), LocalizedDisplayName("strThemeNameSearchBoxBackgroundColor"), LocalizedDescription("strThemeDescriptionSearchBoxBackgroundColor")]
		public Color SearchBoxBackgroundColor {
			get { return _searchBoxBackgroundColor; }
			set {
				if (_searchBoxBackgroundColor == value | value.A < 255)
					return;
				_searchBoxBackgroundColor = value;
				NotifyPropertyChanged("SearchBoxBackgroundColor");
			}
		}

		private Color _searchBoxTextPromptColor = SystemColors.GrayText;
		[LocalizedCategory("strThemeCategoryConnectionsPanel", 2), LocalizedDisplayName("strThemeNameSearchBoxTextPromptColor"), LocalizedDescription("strThemeDescriptionSearchBoxTextPromptColor")]
		public Color SearchBoxTextPromptColor {
			get { return _searchBoxTextPromptColor; }
			set {
				if (_searchBoxTextPromptColor == value)
					return;
				_searchBoxTextPromptColor = value;
				NotifyPropertyChanged("SearchBoxTextPromptColor");
			}
		}

		private Color _searchBoxTextColor = SystemColors.WindowText;
		[LocalizedCategory("strThemeCategoryConnectionsPanel", 2), LocalizedDisplayName("strThemeNameSearchBoxTextColor"), LocalizedDescription("strThemeDescriptionSearchBoxTextColor")]
		public Color SearchBoxTextColor {
			get { return _searchBoxTextColor; }
			set {
				if (_searchBoxTextColor == value)
					return;
				_searchBoxTextColor = value;
				NotifyPropertyChanged("SearchBoxTextColor");
			}
		}
		#endregion

		#region "Config Panel"
		private Color _configPanelBackgroundColor = SystemColors.Window;
		[LocalizedCategory("strThemeCategoryConfigPanel", 3), LocalizedDisplayName("strThemeNameConfigPanelBackgroundColor"), LocalizedDescription("strThemeDescriptionConfigPanelBackgroundColor")]
		public Color ConfigPanelBackgroundColor {
			get { return _configPanelBackgroundColor; }
			set {
				if (_configPanelBackgroundColor == value | value.A < 255)
					return;
				_configPanelBackgroundColor = value;
				NotifyPropertyChanged("ConfigPanelBackgroundColor");
			}
		}

		private Color _configPanelTextColor = SystemColors.WindowText;
		[LocalizedCategory("strThemeCategoryConfigPanel", 3), LocalizedDisplayName("strThemeNameConfigPanelTextColor"), LocalizedDescription("strThemeDescriptionConfigPanelTextColor")]
		public Color ConfigPanelTextColor {
			get { return _configPanelTextColor; }
			set {
				if (_configPanelTextColor == value)
					return;
				_configPanelTextColor = value;
				NotifyPropertyChanged("ConfigPanelTextColor");
			}
		}

		private Color _configPanelCategoryTextColor = SystemColors.ControlText;
		[LocalizedCategory("strThemeCategoryConfigPanel", 3), LocalizedDisplayName("strThemeNameConfigPanelCategoryTextColor"), LocalizedDescription("strThemeDescriptionConfigPanelCategoryTextColor")]
		public Color ConfigPanelCategoryTextColor {
			get { return _configPanelCategoryTextColor; }
			set {
				if (_configPanelCategoryTextColor == value)
					return;
				_configPanelCategoryTextColor = value;
				NotifyPropertyChanged("ConfigPanelCategoryTextColor");
			}
		}

		private Color _configPanelHelpBackgroundColor = SystemColors.Control;
		[LocalizedCategory("strThemeCategoryConfigPanel", 3), LocalizedDisplayName("strThemeNameConfigPanelHelpBackgroundColor"), LocalizedDescription("strThemeDescriptionConfigPanelHelpBackgroundColor")]
		public Color ConfigPanelHelpBackgroundColor {
			get { return _configPanelHelpBackgroundColor; }
			set {
				if (_configPanelHelpBackgroundColor == value | value.A < 255)
					return;
				_configPanelHelpBackgroundColor = value;
				NotifyPropertyChanged("ConfigPanelHelpBackgroundColor");
			}
		}

		private Color _configPanelHelpTextColor = SystemColors.ControlText;
		[LocalizedCategory("strThemeCategoryConfigPanel", 3), LocalizedDisplayName("strThemeNameConfigPanelHelpTextColor"), LocalizedDescription("strThemeDescriptionConfigPanelHelpTextColor")]
		public Color ConfigPanelHelpTextColor {
			get { return _configPanelHelpTextColor; }
			set {
				if (_configPanelHelpTextColor == value)
					return;
				_configPanelHelpTextColor = value;
				NotifyPropertyChanged("ConfigPanelHelpTextColor");
			}
		}

		private Color _configPanelGridLineColor = SystemColors.InactiveBorder;
		[LocalizedCategory("strThemeCategoryConfigPanel", 3), LocalizedDisplayName("strThemeNameConfigPanelGridLineColor"), LocalizedDescription("strThemeDescriptionConfigPanelGridLineColor")]
		public Color ConfigPanelGridLineColor {
			get { return _configPanelGridLineColor; }
			set {
				if (_configPanelGridLineColor == value)
					return;
				_configPanelGridLineColor = value;
				NotifyPropertyChanged("ConfigPanelGridLineColor");
			}
		}
		#endregion
		#endregion
	}
}
